### ISSUE TITLE
.flat / .flatMap Rule Duplication

### ISSUE THREAD

**Issue by karlhorky** [EvgenyOrekhov/eslint-config-hardcore#435]
First of all, thanks for this config! Learned about a lot of plugins 🙌

It seems like there may be some duplication of rules in the base config (see below). But not sure if this is something important to this project - if not, feel free to close :)

```
    "unicorn/prefer-array-flat-map": "error",
    "unicorn/prefer-array-flat": "error",
    "array-func/prefer-flat-map": "error",
    "array-func/prefer-flat": "error",
```

**Comment by EvgenyOrekhov** [comment]
You're welcome!

Yeah I believe there are a few more duplicated rules besides those. The thing is, one rule might catch some cases that the other rule can't, and vice versa. So some of those duplicated rules actually complement each other. I think this benefit is more important than the fact that sometimes you will get duplicated warnings. So I think I should keep duplicated rules enabled to cover more cases. Do you think it makes sense?

**Comment by EvgenyOrekhov** [comment]
As another option, we could ask the maintainers of eslint-plugin-array-func and eslint-plugin-unicorn to join forces and "merge" their rules into eslint-plugin-unicorn. I actually did this with regex-related plugins, see https://github.com/RunDevelopment/eslint-plugin-clean-regex/issues/28.

**Comment by karlhorky** [comment]
> The thing is, one rule might catch some cases that the other rule can't, and vice versa. So some of those duplicated rules actually complement each other. I think this benefit is more important than the fact that sometimes you will get duplicated warnings. So I think I should keep duplicated rules enabled to cover more cases. Do you think it makes sense?

Yep, makes sense! Then I'll close this one.

**Comment by karlhorky** [comment]
> As another option, we could ask the maintainers of eslint-plugin-array-func and eslint-plugin-unicorn to join forces and "merge" their rules into eslint-plugin-unicorn.

This is true! That would be cool 👍 Sometimes maintainers are open to things like this.

**Comment by EvgenyOrekhov** [comment]
@karlhorky I reviewed those `.flat()` / `.flatMap()` rules. Unicorn's rules cover more cases, so I disabled array-func's rules. Will be released in the next version.

**Comment by EvgenyOrekhov** [comment]
Released in [v24.0.0](https://github.com/EvgenyOrekhov/eslint-config-hardcore/releases/tag/v24.0.0).
